pkg/monitors/grafana/grafana-monitor.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		return nil, fmt.Errorf("Error listing probes %v", err)
+	}
+
+	probeIDs := make([]int64, len(probes))
+	for i, p := range probes {
+		probeIDs[i] = p.Id
+	}
+
+	var checkId int64
+	if len(monitor.ID) > 0 {
+		idResult, err := strconv.ParseInt(monitor.ID, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("Error converting ID %v %v", monitor.ID, err)
+		}
+		checkId = idResult
+	}
+	var tentantId int64
+	grafanaConfig, _ := monitor.Config.(*endpointmonitorv1alpha1.GrafanaConfig)
+	if grafanaConfig != nil {
+		tentantId = grafanaConfig.TenantId
+	}
+	// Creating a new Check object
+	return &synthetic_monitoring.Check{
+		Id:        checkId,
+		Target:    monitor.URL,
+		Job:       monitor.Name,
+		Frequency: service.frequency,
+		TenantId:  tentantId,
+		Timeout:   2000,
+		Enabled:   true,
+		Probes:    probeIDs,
+		Settings: synthetic_monitoring.CheckSettings{
+			Http: &synthetic_monitoring.HttpSettings{
+				IpVersion: synthetic_monitoring.IpVersion_V4,
+			},
+		},
+		BasicMetricsOnly: true,
+	}, nil
+}
+
+// Add adds a new monitor to Grafana Synthetic Monitoring service
+func (service *GrafanaMonitorService) Add(monitor models.Monitor) {
+	newCheck, err := service.CreateSyntheticCheck(monitor)
+	if err != nil {
+		log.Error(err, "Failed to create synthetic check")
+		return
+	}
+
+	// Using the synthetic monitoring client to add the new check
+	createdCheck, err := service.smClient.AddCheck(service.ctx, *newCheck)
+	if err != nil {
+		log.Error(err, "Failed to add new monitor")
+		return
+	}
+
+	log.Info(fmt.Sprintf("Successfully added new monitor %v %v", monitor.ID, createdCheck.Id))
+}
+
+func (service *GrafanaMonitorService) Update(monitor models.Monitor) {
+	newCheck, err := service.CreateSyntheticCheck(monitor)
+	if err != nil {
+		log.Error(err, "Failed to create synthetic check")
+		return
+	}
+	// Using the synthetic monitoring client to update the old check
+	createdCheck, err := service.smClient.UpdateCheck(service.ctx, *newCheck)
+	if err != nil {
+		log.Error(err, "Failed to update monitor")
+		return
+	}
+
+	log.Info(fmt.Sprintf("Successfully updated monitor %v %v", monitor.ID, createdCheck.Id))
+}
+
+func (service *GrafanaMonitorService) GetByName(name string) (*models.Monitor, error) {
+	monitors := service.GetAll()
+	for _, m := range monitors {
+		if m.Name == name {
+			return &m, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Unable to locate monitor with name %v", name)
+}
+
+func (service *GrafanaMonitorService) Remove(monitor models.Monitor) {
+	// Convert string to base64 int
+	Id, err := strconv.ParseInt(monitor.ID, 10, 64)
+	if err != nil {
+		log.Info("Failed to parse int", monitor.ID)
+		return
+	}
+	err = service.smClient.DeleteCheck(service.ctx, Id)
+	if err != nil {
+		log.Error(err, "Failed to delete monitor")
+		return
+	}
+
+}
+
+func (service *GrafanaMonitorService) Equal(oldMonitor models.Monitor, newMonitor models.Monitor) bool {
+	// TODO Implement Deep equal for config as well
```

Comment:
Is this TODO talking about something like:

reflect.DeepEqual(oldconfig, newconfig) 

if yes, any reason to not do this right now and keep it as a TODO?

Reply:
I was unsure about if it was something more, for instance fetching from remote or similar, but I forgot that I added a TODO. Will fix :heavy_check_mark: